### PR TITLE
Cleaned up some of the code and sanitize the sentence prior to matching

### LINKED ISSUE #798: Sentence beginning breaks in case of enumerating symbols
When items of an enumeration are preceded by enumerating symbols (instead of`<li>`), the sentence beginnings assessments views these symbols as the first word of these sentences, resulting in a red bullet.

Do we want to support this non-HTML? 

## Input
```
<h3>Weitere Tipps zum Vorstellungsgespräch</h3>
<p>✔ <a href="http://karrierebibel.de/das-vorstellungsgesprach/" target="_blank">Vorstellungsgespräch</a><br />
✔ <a href="http://karrierebibel.de/bewerbungsgesprach-vorbereiten-gesprachsphasen/" target="_blank">5 typische Gesprächsphasen</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgesprach-vorbereitung/" target="_blank">Vorstellungsgespräch vorbereiten</a><br />
✔ <a href="http://karrierebibel.de/dresscode-im-vorstellungsgespraech/" target="_blank">Dresscode</a><br />
✔ <a href="http://karrierebibel.de/selbstvorstellung-im-bewerbungsgespraech/" target="_blank">Begrüßungstipps</a><br />
✔ <a href="http://karrierebibel.de/selbstpraesentation/" target="_blank">Selbstpräsentation</a><br />
✔ <a href="http://karrierebibel.de/gesprachsfuhrung-bewerbung/" target="_blank">Gesprächsführung</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgespraech-beenden/" target="_blank">Vorstellungsgespräch beenden</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgespraech-nervositaet/" target="_blank">Nervosität überwinden</a><br />
✔ <a href="http://karrierebibel.de/authentizitat-die-kunst-authentisch-zu-sein-und-zu-bleiben/" target="_blank">Die Kunst, authentisch zu sein</a></p>
<h4>Vorstellungsgespräch: Fragen und Antworten</h4>
<p>✔ <a href="http://karrierebibel.de/vorstellungsgesprach-fragen/" target="_blank">100 Fragen an Bewerber</a><br />
✔ <a href="http://karrierebibel.de/fangfragen-im-bewerbungsgespraech/" target="_blank">25 Fangfragen</a><br />
✔ <a href="http://karrierebibel.de/bewerberfrage-wie-hoch-war-ihr-letztes-gehalt/" target="_blank">Wie hoch war Ihr letztes Gehalt?</a><br />
✔ <a href="http://karrierebibel.de/bewerberfrage-wo-sehen-sie-sich-in-5-jahren/" target="_blank">Wo sehen Sie sich in 5 Jahren?</a><br />
✔ <a href="http://karrierebibel.de/bewerberfrage-warum-haben-sie-gekundigt/" target="_blank">Warum haben Sie gekündigt?</a><br />
✔ <a href="http://karrierebibel.de/bewerberfrage-haben-sie-sich-woanders-beworben/" target="_blank">Haben Sie sich woanders beworben?</a><br />
✔ <a href="http://karrierebibel.de/bewerberfrage-was-waeren-sie-fuer-ein-tier/" target="_blank">Was wären Sie für ein Tier?</a><br />
✔ <a href="http://karrierebibel.de/warum-sollten-wir-sie-einstellen/" target="_blank">Warum sollten wir Sie einstellen?</a><br />
✔ <a href="http://karrierebibel.de/bewerberfrage-wie-stehen-sie-zu-ueberstunden/" target="_blank">Wie stehen Sie zu Überstunden?</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgespraech-staerken-schwaechen/" target="_blank">Was sind Ihre Schwächen?</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgespraech-checkliste-unzulassige-fragen/" target="_blank">Unzulässige Fragen</a><br />
✔ <a href="http://karrierebibel.de/ruckfragen-bewerbungsgesprach/" target="_blank">Rückfragen an Personaler</a></p>
<h4>Bewerbungsgespräch Tipps & Tricks</h4>
<p>✔ <a href="http://karrierebibel.de/jobinterview-umfeld/" target="_blank">Achten Sie aufs Umfeld</a><br />
✔ <a href="http://karrierebibel.de/bewerbungsgespraech-fehler/" target="_blank">Bewerbungsgespräch Fehler</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgespraech-koerperhaltung/" target="_blank">Körpersprache Tipps</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgesprach-auf-englisch/" target="_blank">Englisch im Vorstellungsgespräch</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgespraech-abbrechen/" target="_blank">Vorstellungsgespräch abbrechen</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgesprach-gut-gelaufen/" target="_blank">Vorstellungsgespräch gut gelaufen?</a><br />
✔ <a href="http://karrierebibel.de/nachfassen-nach-vorstellungsgespraech/" target="_blank">Nachfassen</a><br />
✔ <a href="http://karrierebibel.de/zweites-vorstellungsgespraech/" target="_blank">Zweites Vorstellungsgespräch</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgesprach-fahrtkostenerstattung/" target="_blank">Reisekosten: Wer zahlt?</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgespraech-telefon-video-skype/" target="_blank">Bewerbungsgespräch per Video</a><br />
✔ <a href="http://karrierebibel.de/telefoninterview-vorstellungsgesprach-tipps-fragen/" target="_blank">Telefoninterview Tipps</a><br />
✔ <a href="http://karrierebibel.de/assessment-center/" target="_blank">Assessment Center</a><br />
✔ <a href="http://karrierebibel.de/prufungsangst-uberwinden-tipps/" target="_blank">Tipps gegen Prüfungsangst</a><br />
✔ <a href="http://karrierebibel.de/headhunter-jobs/" target="_blank">Anruf vom Headhunter</a><br />
✔ <a href="http://karrierebibel.de/outplacement-beratung/" target="_blank">Outplacement Beratung</a><br />
✔ <a href="http://karrierebibel.de/vorstellungsgespraech-trainieren/" target="_blank">Vorstellungsgespräch trainieren</a></p>
```

### PR DESCRIPTION
**Pull request by jcomack**
This strips off any non-alphanumeric characters off the beginning of a sentence prior to matching sentence beginnings.

Fixes #798

### CODE REVIEW

File: js/researches/getSentenceBeginnings.js [hunk 1]
```diff
@@ -51,25 +52,56 @@ var compareFirstWords = function ( sentenceBeginnings, sentences ) {
...
 
 /**
+ * Sanitizes the sentence.
+ *
+ * @param {string} sentence The sentence to sanitize.
+ * @returns {string} The sanitized sentence.
+ */
+function sanitizeSentence( sentence ) {
+	sentence = stripTags( sentence );
+	sentence = 	sentence.replace( /^[^A-Za-z0-9]/, "" );
```
Comment (terw-dan) [inline js/researches/getSentenceBeginnings.js]: Is there an extra tab here? Or is it just github?

**Comment by terw-dan** [comment]
CR done :+1:

**Comment by boblinthorst** [comment]
Acceptance: ok 👍

File: js/researches/getSentenceBeginnings.js [hunk 2]
```diff
@@ -51,25 +52,56 @@ var compareFirstWords = function ( sentenceBeginnings, sentences ) {
...
+}
+
+/**
+ * Retrieves the first word from the sentence.
+ *
+ * @param {string} sentence The sentence to retrieve the first word from.
+ * @param {Array} firstWordExceptions Exceptions to match against.
+ * @returns {string} The first word of the sentence.
+ */
+function getFirstWord( sentence, firstWordExceptions ) {
```
Comment (atimmer) [inline js/researches/getSentenceBeginnings.js]: Does this actually retrieve the first word of the sentence or does it do more?

**Pull request merged**